src/main/java/org/folio/rest/service/action/validation/DefaultActionValidationService.java

```diff
@@ -40,8 +40,10 @@ protected void validateAccountStatus(Account account) {
   }
 
   @Override
-  protected MonetaryValue calculateRemainingBalance(Account account, MonetaryValue requestedAmount) {
-    return new MonetaryValue(account.getRemaining()).subtract(requestedAmount);
+  protected Future<MonetaryValue> calculateRemainingBalance(Account account,
```

Comment:
Do we really need to "futurize" this method?

Reply:
Yes, calculation of `remainingAmount` is based on calculation of refundable amount, which requires calling the DB, thus making the whole process asynchronous.